examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -33,6 +33,20 @@ using namespace ::chip::Controller;
 CHIP_ERROR PairingCommand::RunCommand()
 {
     CurrentCommissioner().RegisterPairingDelegate(this);
+    if (mCASEAuthTags.HasValue() && mCASEAuthTags.Value().size() <= kMaxSubjectCATAttributeCount)
+    {
+        CATValues cats = kUndefinedCATs;
+        for (size_t index = 0; index < mCASEAuthTags.Value().size(); ++index)
+        {
+            cats.values[index] = mCASEAuthTags.Value()[index];
+        }
+        if (cats.AreValid())
+        {
+            ExampleOperationalCredentialsIssuer * opIssuer =
+                reinterpret_cast<chip::Controller::ExampleOperationalCredentialsIssuer *>(mCredIssuerCmds->GetCredentialIssuer());
```

Comment:
Please just fix `GetCredentialIssuer` to return `ExampleOperationalCredentialsIssuer *`, if that's what it always has to return.

Reply:
So change the virtual function `GetCredentialIssuer()` to return `ExampleOperationalCredentialsIssuer *`?`mCredIssuerCmds` is `CredentialIssuerCommands` instead of `ExampleCredentialIssuerCommands`. Should we add a function returning an example-implement member variable to a virtual class?

Reply:
Ah, I see.  Hmm.  So I guess the point here is that `CredentialIssuerCommands` can be plugged in with another implementation.

In that case, what we should do is add a function on `CredentialIssuerCommands` to set the CATs, and the example impl can pass it on to its example credentials issuer, and other impls would do whatever they want.

Reply:
Done